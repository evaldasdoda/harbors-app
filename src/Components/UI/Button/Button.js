import React from 'react';
import PropTypes from 'prop-types';

// Styles
require('./Button.scss');

const button = props => {
    return (
        <button disabled={props.disabled} className={'BUTTON btn ' + props.class} onClick={props.clicked} type={props.type}>
            {props.children}
        </button>
    );
};

button.propTypes = {
    disabled: PropTypes.bool,
    class: PropTypes.string,
    onClick: PropTypes.func,
    type: PropTypes.string
};

export default button;
