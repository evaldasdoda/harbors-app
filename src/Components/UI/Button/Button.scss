@import "../../../Assets/styles/setup";

.BUTTON {
    & a{
        color: $white;
        text-decoration: none;
    }
    &:active,
    &:hover {
    }
    &-rev {
    }
    &-active {
    }
}
