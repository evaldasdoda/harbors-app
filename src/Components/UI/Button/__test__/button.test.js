import React from 'react';
import ReactDOM from 'react-dom';
import Button from '../Button';

import {render, cleanup} from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import renderer from 'react-test-renderer';

afterEach(cleanup)

it("renders without crashing", ()=>{
    const div = document.createElement("div");
    ReactDOM.render(<Button></Button>, div)
})

it("renders button correctly",()=>{
    const {getByTestId} = render(<Button type="button" disabled={false} onClick={jest.fn}>Submit</Button>)
    expect(getByTestId('button')).toHaveTextContent('Submit')
})

it("renders button correctly",()=>{
    const {getByTestId} = render(<Button type="button" disabled={false}>Go to map</Button>)
    expect(getByTestId('button')).toHaveTextContent('Go to map')
})

it("matches snapshot",()=>{
    const tree = renderer.create(
        <Button 
            type="button" 
            disabled={false} 
            onClick={jest.fn}
            class="btn-danger" >Submit
        </Button>).toJSON();
    expect(tree).toMatchSnapshot();
})