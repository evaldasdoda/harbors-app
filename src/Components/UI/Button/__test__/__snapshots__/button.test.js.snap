// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="BUTTON btn btn-danger"
  data-testid="button"
  disabled={false}
  type="button"
>
  Submit
</button>
`;
